executable-monitor/executable-monitor.py

```diff
@@ -72,76 +81,114 @@
     logging.info(f"Running executable: {exe_abs_path} ")
     logging.info(f"Storing logs in: {log_dir}")
     logging.info(f"Timeout (seconds): {args.timeout_seconds}")
+    logging.info(f"Searching for success line: {args.success_line}")
+    logging.info(f"Will re-try the run {retryAttempts} times")
+    if args.success_exit_status is not None:
+        logging.info("Looking for exit status {0}".format(args.success_exit_status ))
+    for attempts in range(0,retryAttempts + 1):
+
+        # Initialize values
+        success_line = ""
+        timeout_occurred = False
+        exe_exit_status = None
+        exe_exitted = False
+        success_line_found = False
+        exit_condition_met = False
+        wait_for_exit = args.success_exit_status is not None
+
+        # Create two file descriptors. The subprocess writes to one, the parent task reads from the other
+        # This is a workaround to avoid the fact that calling readline() on the stdout of the subprocess is
+        # a blocking call. Where if the subproces is running but hasn't printed anything, readline will never time out.
+        # The approach uses the underlying file system to rely on an end of file character to ensure that it does not hang.
+        WriteOutputFile = open("output.log", "w")
+        ReadOutputFile = open("output.log", "r")
+
+        # Launch the executable
+        exe = subprocess.Popen([exe_abs_path], stdout=WriteOutputFile, stderr=WriteOutputFile, universal_newlines=True)
```

Comment:
This is clobbering the environment passed in, which makes it tough to debug which dynamic libraries are in use.